persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/connectionfactory/PooledConnectionFactory.java

```diff
@@ -14,86 +25,162 @@
+            hikariPropPath = hikariUrl != null ? hikariUrl.getPath() : null;
+         }
+      }
+
+      if (forceC3P0 || hikariPropPath == null) {
+         URL c3p0Props = fileLookup.lookupFileLocation(C3P0_PROPERTIES, classLoader);
+         URL c3p0Xml = fileLookup.lookupFileLocation(C3P0_CONFIG, classLoader);
+         if (LOG.isDebugEnabled()) {
+            if (c3p0Props != null)
+               LOG.debugf("Found '%s' in classpath: %s", C3P0_PROPERTIES, c3p0Props);
+            if (c3p0Xml != null)
+               LOG.debugf("Found '%s' in classpath: %s", C3P0_CONFIG, c3p0Xml);
+         }
+
+         // If C3PO is not forced and no C3P0 properties are found, then do nothing so HikariCP is initialised
+         if (forceC3P0 || c3p0Props != null || c3p0Xml != null) {
+            hikariPoolEnabled = false;
+            initC3P0DataSource(poolConfig);
+            return;
+         }
+      }
+      initHikariDataSource(hikariPropPath, poolConfig);
+   }
+
+   private void initHikariDataSource(String propertiesPath, PooledConnectionFactoryConfiguration poolConfig) {
+      try {
+         HikariConfig hikariConfig = propertiesPath != null ? new HikariConfig(propertiesPath) : new HikariConfig();
+         hikariConfig.setRegisterMbeans(true);
+
+         // Set Hikari properties from poolConfig only if there are no existing values specified in the properties file
+         if (hikariConfig.getDataSourceClassName() == null && hikariConfig.getJdbcUrl() == null)
+            hikariConfig.setJdbcUrl(poolConfig.connectionUrl());
+         if (hikariConfig.getUsername() == null)
+            hikariConfig.setUsername(poolConfig.username());
+         if (hikariConfig.getPassword() == null)
+            hikariConfig.setPassword(poolConfig.password());
+         if (hikariConfig.getDriverClassName() == null)
+            hikariConfig.setDriverClassName(poolConfig.driverClass());
+
+         hikari = new HikariDataSource(hikariConfig);
+      } catch (Exception e) {
+         // Exception should only be thrown when trying to setDriverClassName either explicitly or as part of HikariConfig
+         LOG.errorInstantiatingJdbcDriver(poolConfig.driverClass(), e);
+         throw new PersistenceException(String.format(
+               "Error while instantiating JDBC driver: '%s'", poolConfig.driverClass()), e);
       }
-      pooledDataSource = new ComboPooledDataSource();
-      pooledDataSource.setProperties(new Properties());
+
+      try {
+         MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+         ObjectName poolName = new ObjectName("com.zaxxer.hikari:type=Pool (" + hikari.getPoolName() + ")");
+         hikariMxBean = JMX.newMXBeanProxy(mBeanServer, poolName, HikariPoolMXBean.class);
+      } catch (MalformedObjectNameException ignore) {
+         // Ignore as this should never happen
+      }
+   }
+
+   private void initC3P0DataSource(PooledConnectionFactoryConfiguration pooledConfiguration) {
```

Comment:
Having both `initC3P0DataSource` and `initHikariDataSource` is a sign that we need 2 classes...